contracts/exchange/test/utils/match_order_tester.ts

```diff
@@ -126,6 +148,59 @@ export class MatchOrderTester {
         this._initialTokenBalancesPromise = this.getBalancesAsync();
     }
 
+    /**
+     * Performs batch order matching on a set of complementary orders and asserts results.
+     * @param orders The list of orders and filled states
+     * @param matchPairs An array of left and right indices that will be used to perform
+     *                   the expected simulation.
+     * @param takerAddress Address of taker (the address who matched the two orders)
+     * @param expectedTransferAmounts Expected amounts transferred as a result of each round of
+     *                                order matching. Omitted fields are either set to 0 or their
+     *                                complementary field.
+     * @return Results of `batchMatchOrders()`.
+     */
+    public async batchMatchOrdersAndAssertEffectsAsync(
+        orders: BatchMatchedOrders,
+        takerAddress: string,
+        matchPairs: Array<[number, number]>,
+        expectedTransferAmounts: Array<Partial<MatchTransferAmounts>>,
+        initialTokenBalances?: TokenBalances,
+    ): Promise<BatchMatchResults> {
+        // Ensure that the provided input is valid.
+        expect(matchPairs.length).to.be.eq(expectedTransferAmounts.length);
+        expect(orders.leftOrders.length).to.be.eq(orders.leftOrdersTakerAssetFilledAmounts.length);
+        expect(orders.rightOrders.length).to.be.eq(orders.rightOrdersTakerAssetFilledAmounts.length);
+        // Ensure that the exchange is in the expected state.
+        await assertBatchOrderStatesAsync(orders, this.exchangeWrapper);
+        // Get the token balances before executing `batchMatchOrders()`.
+        const _initialTokenBalances = initialTokenBalances
+            ? initialTokenBalances
+            : await this._initialTokenBalancesPromise;
+        // Execute `batchMatchOrders()`
```

Comment:
Right before this, can we do an `exchangeWrapper.getBatchMatchOrdersResultsAsync()` and assert the results against the simulation? I don't see where we are checking the return value of `batchMatchOrders()` anywhere.

Would also be nice to have this done for the regular `matchOrders()` tests, as well.